src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -99,6 +99,7 @@ def is_hive(self) -> bool:
 
     @property
     def key(self) -> str:
+        # TODO: figure out what the difference is between key and full_name
```

Comment:
@nfx, @william-conti and @ericvergnaud : (Tagging you as you had the most recent changes with this code and the `full_name` attribute.) I would like the code to be more consistent with Databricks naming conventions so that it is easier for (new) developers to understand the intent. 

As I see it:
- `key` should be `table_reference`, see [this](https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-qry-select-table-reference.html). Or maybe we call it `sql_table_reference` to make clear it is intended to be used in Databricks SQL queries.
   - Synonym for `full_name` in most cases. 
   - Possible a reference to a location, as shown in the code.
- `full_name` aligns with Databricks SDK

Reply:
we can change `key` to `full_name`, but in a separate PR

Reply:
https://github.com/databrickslabs/ucx/issues/2979